src/Type/Schema.php

```diff
@@ -339,15 +341,22 @@ private function loadType(string $typeName) : ?Type
 
         $type = $typeLoader($typeName);
 
+        if (is_callable($type)) {
+            $type = self::resolveType($type);
+        }
+
         if (! $type instanceof Type) {
```

Comment:
```suggestion
        $type = self::resolveType($type);
```

That line is all we need. No additional checks for `is_callable()`, no checks if `$type` is indeed a type.

Reply:
Can't do it without breaking a lot of tests. 😢 Will revisit in next PR.

Reply:
GitHub won't let me PR from this repo to your branch, but here is how i propose to fix this: https://github.com/webonyx/graphql-php/commit/a705ad46c83fdcc866448d707adb92f45527ea0a

Reply:
Sure, looks lovely, but the last time I messed with the tests I got push-back from @vladar . So if it's okay with him, it's okay with me, but I don't want to do anything until I hear it from him.

Reply:
Most importantly, we should be consistent and always throw either a `TypeError` or a crafted `InvariantViolation`. As of now, this PR does a mix of both, depending on whether the type was lazy or not.

My opinion is that we can let the errors happen right at PHP's type boundaries. In contrast to JavaScript, PHP gives us those checks essentially for free and includes a readable error message and a stacktrace. Since those type errors work the same everywhere, they should be quite easy to understand.

Reply:
I'm well aware of both flavors. My goal is to get this PR merged with minimal impact to the existing tests, because once we start removing all the "invariant" stuff then we start changing the functionality of the current "validate" functionality, and before you know it I'll have lost another 3 months to this PR. For the love of all that is holy, please just let it rest for right now and I swear I will revisit all of this stuff in my next PR and we can all have a great time arguing about this stuff then.

Reply:
```suggestion
        if (is_callable($type)) {
            $type = $type();
        }

        if (! $type instanceof Type) {
```

Let's go with this then. That way, we throw a consistent error.

Reply:
If it will satisfy you, then fine: 6a1fad4